wagtail/wagtailsnippets/tests.py

```diff
@@ -90,6 +92,20 @@ def test_create(self):
         self.assertEqual(snippets.count(), 1)
         self.assertEqual(snippets.first().url, 'http://www.example.com/')
 
+    def test_create_with_tags(self):
+        tags = ['hello', 'world']
+        response = self.post(post_data={'text': 'test_advert',
+                                        'url': 'http://example.com/',
+                                        'tags': ', '.join(tags)})
+
+        self.assertRedirects(response, reverse('wagtailsnippets:list',
+                                               args=('tests', 'advert')))
+
+        snippet = Advert.objects.get(text='test_advert')
+        self.assertEqual(
+            list(snippet.tags.order_by('name')),
+            list(Tag.objects.order_by('name').filter(name__in=tags)))
```

Comment:
We should also test that these lists are non-empty (since this assertion would also pass if the tags field was ignored completely).